src/generator/macro.jl

```diff
@@ -66,7 +66,11 @@ function c_non_float_literal_to_julia(sfx)
     elseif occursin("ul", sfx) || occursin("lu", sfx)
         return "Culong"
     elseif !unsigned && endswith(sfx, "l")
-        return "Clong"
+        if length(txt_no_sfx) == 10 && startswith(txt_no_sfx, r"0x[8-9a-fA-F]")
```

Comment:
I think it might be simpler to try parsing the string as a `Clong` and falling back to `Culong` if that fails.

Reply:
True, it would also help with non-hexadecimal representations that may still represent large numbers.